tracker/btn.go

```diff
@@ -0,0 +1,168 @@
+package tracker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/autobrr/tqm/httputils"
+	"github.com/autobrr/tqm/logger"
+	"github.com/sirupsen/logrus"
+	"go.uber.org/ratelimit"
+)
+
+type BTNConfig struct {
+	Key string `koanf:"api_key"`
+}
+
+type BTN struct {
+	cfg  BTNConfig
+	http *http.Client
+	log  *logrus.Entry
+}
+
+func NewBTN(c BTNConfig) *BTN {
+	l := logger.GetLogger("btn-api")
+	return &BTN{
+		cfg:  c,
+		http: httputils.NewRetryableHttpClient(15*time.Second, ratelimit.New(1, ratelimit.WithoutSlack), l),
+		log:  l,
+	}
+}
+
+func (c *BTN) Name() string {
+	return "BTN"
+}
+
+func (c *BTN) Check(host string) bool {
+	return strings.EqualFold(host, "landof.tv")
+}
+
+// extractTorrentID extracts the torrent ID from the torrent comment field
+func (c *BTN) extractTorrentID(comment string) (string, error) {
+	if comment == "" {
+		return "", fmt.Errorf("empty comment field")
+	}
+
+	re := regexp.MustCompile(`https?://[^/]*broadcasthe\.net/torrents\.php\?action=reqlink&id=(\d+)`)
```

Comment:
This could be set as a const outside of the struct to only have to compile the regex once per run.

Reply:
Fixed!